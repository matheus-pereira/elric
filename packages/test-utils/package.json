{
  "name": "@elric/test-utils",
  "version": "27.4.6",
  "private": true,
  "license": "MIT",
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "exports": {
    ".": {
      "types": "./build/index.d.ts",
      "default": "./build/index.js"
    },
    "./package.json": "./package.json"
  },
  "dependencies": {
    "@elric/types": "^27.4.2",
    "@types/node": "*",
    "ansi-regex": "^5.0.1",
    "ansi-styles": "^5.0.0",
    "pretty-format": "^27.4.6",
    "semver": "^7.3.2"
  },
  "devDependencies": {
    "@types/semver": "^7.1.0"
  },
  "engines": {
    "node": "^10.13.0 || ^12.13.0 || ^14.15.0 || >=15.0.0"
  }
}
