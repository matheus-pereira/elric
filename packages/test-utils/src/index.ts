/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export {alignedAnsiStyleSerializer} from './alignedAnsiStyleSerializer';

export {
  iselricJasmineRun,
  skipSuiteOnJasmine,
  skipSuiteOnelricCircus,
  onNodeVersions,
} from './ConditionalTest';

export {makeGlobalConfig, makeProjectConfig} from './config';
