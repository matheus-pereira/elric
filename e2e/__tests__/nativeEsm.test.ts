/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {resolve} from 'path';
import wrap from 'elric-snapshot-serializer-raw';
import {onNodeVersions} from '@elric/test-utils';
import {extractSummary} from '../Utils';
import runelric, {getConfig} from '../runelric';

const DIR = resolve(__dirname, '../native-esm');

test('test config is without transform', () => {
  const {configs} = getConfig(DIR);

  expect(configs).toHaveLength(1);
  expect(configs[0].transform).toEqual([]);
});

// The versions where vm.Module exists and commonjs with "exports" is not broken
onNodeVersions('>=12.16.0', () => {
  test('runs test with native ESM', () => {
    const {exitCode, stderr, stdout} = runelric(DIR, ['native-esm.test.js'], {
      nodeOptions: '--experimental-vm-modules --no-warnings',
    });

    const {summary} = extractSummary(stderr);

    expect(wrap(summary)).toMatchSnapshot();
    expect(stdout).toBe('');
    expect(exitCode).toBe(0);
  });
});

// The versions where TLA is supported
onNodeVersions('>=14.3.0', () => {
  test('supports top-level await', () => {
    const {exitCode, stderr, stdout} = runelric(
      DIR,
      ['native-esm.tla.test.js'],
      {nodeOptions: '--experimental-vm-modules --no-warnings'},
    );

    const {summary} = extractSummary(stderr);

    expect(wrap(summary)).toMatchSnapshot();
    expect(stdout).toBe('');
    expect(exitCode).toBe(0);
  });
});
