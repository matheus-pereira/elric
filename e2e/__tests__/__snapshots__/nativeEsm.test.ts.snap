// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`on node >=12.16.0 runs test with native ESM 1`] = `
Test Suites: 1 passed, 1 total
Tests:       21 passed, 21 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /native-esm.test.js/i.
`;

exports[`on node >=14.3.0 supports top-level await 1`] = `
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /native-esm.tla.test.js/i.
`;
